src/System.cpp

```diff
@@ -32,7 +32,7 @@
 #include <stdexcept>
 #endif // _WIN32
 
-#if defined(__MINGW32__)
+#if defined(__MINGW32__) && (__GNUC__ < 6)
```

Comment:
Maybe better: `#if defined(__MINGW32__) && !defined(REFKNOWNFOLDERID )` ?
I'm a bit hesitant to rely on 2 compiler variables.

Reply:
we could try that yes